alerting/src/main/kotlin/org/opensearch/alerting/util/SupportedApiSettingsExtensions.kt

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.alerting.util
+
+import org.opensearch.action.ActionResponse
+import org.opensearch.action.admin.cluster.health.ClusterHealthRequest
+import org.opensearch.action.admin.cluster.health.ClusterHealthResponse
+import org.opensearch.action.admin.cluster.node.stats.NodesStatsRequest
+import org.opensearch.action.admin.cluster.node.stats.NodesStatsResponse
+import org.opensearch.action.admin.cluster.node.tasks.list.ListTasksRequest
+import org.opensearch.action.admin.cluster.node.tasks.list.ListTasksResponse
+import org.opensearch.action.admin.cluster.repositories.get.GetRepositoriesRequest
+import org.opensearch.action.admin.cluster.repositories.get.GetRepositoriesResponse
+import org.opensearch.action.admin.cluster.settings.ClusterGetSettingsResponse
+import org.opensearch.action.admin.cluster.snapshots.get.GetSnapshotsRequest
+import org.opensearch.action.admin.cluster.snapshots.get.GetSnapshotsResponse
+import org.opensearch.action.admin.cluster.state.ClusterStateRequest
+import org.opensearch.action.admin.cluster.stats.ClusterStatsRequest
+import org.opensearch.action.admin.cluster.stats.ClusterStatsResponse
+import org.opensearch.action.admin.cluster.tasks.PendingClusterTasksRequest
+import org.opensearch.action.admin.cluster.tasks.PendingClusterTasksResponse
+import org.opensearch.action.admin.indices.recovery.RecoveryRequest
+import org.opensearch.action.admin.indices.recovery.RecoveryResponse
+import org.opensearch.alerting.core.model.ClusterMetricsInput
+import org.opensearch.alerting.core.model.ClusterMetricsInput.ApiType
+import org.opensearch.alerting.elasticapi.convertToMap
+import org.opensearch.alerting.settings.SupportedApiSettings
+import org.opensearch.alerting.settings.SupportedApiSettings.Companion.resolveToActionRequest
+import org.opensearch.client.Client
+import org.opensearch.common.settings.Settings
+import org.opensearch.common.xcontent.support.XContentMapValues
+
```

Comment:
Rename class to use `clusterMetrics` instead of `api`

Reply:
Good point. Adjusted the name accordingly.